dependabot-core.gemspec

```
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+# This is a placeholder gem to prevent namesquatting on https://rubygems.org/gems/dependabot-core
+# Any updates must be manually published to RubyGems.
+# It's excluded from the normal release process because it's not expected to change.
+
+Gem::Specification.new do |spec|
+  spec.name = "dependabot-core"
+  spec.summary      = "This is a placeholder gem to prevent namesquatting. " \
+                      "You probably want the gem 'dependabot-omnibus'."
+  spec.description  = "This is a placeholder gem to prevent namesquatting. " \
+                      "You probably want the gem 'dependabot-omnibus'."
+  spec.authors     = "Dependabot"
+  spec.email       = "opensource@github.com"
+  spec.files       = [] # intentionally empty, this is a placeholder gem to prevent namesquatting
+  spec.homepage    = "https://github.com/dependabot/dependabot-core"
+  spec.license = "Nonstandard" # License Zero Prosperity Public License
+
+  spec.metadata = {
+    "bug_tracker_uri" => "https://github.com/dependabot/dependabot-core/issues"
+  }
+
+  spec.version = "0.95.1"
```

I understand this gemspec will not really do much, but I think it makes sense to point to the current version. Note: 👇 will necessitate adding `require "./lib/dependabot"` to the top of this file.

```suggestion
  spec.version = Dependabot::VERSION
```

---

I don't mind doing this at all, but curious why you think it's an improvement?

And thank you for the Ruby pointer on the `require`!

---

> I'm happy to do this, but curious why you think it's an improvement? Since it's disconnected from the rest of our code... and also won't be published regularly to RubyGems, so the version bumps would be "jarring" perhaps?

> Only benefit I see is being consistent with how we handle versions in the other gems.

My thoughts are that it signals the existence of a newer version. I'm positive you've thought more deeply about this than me, so please feel free to proceed as you see fit 😄

---

Thanks Landon, I appreciate that.

Circling back on this with fresh eyes. I personally think as a drive-by dev, if I see a hardcoded version, and it maps to the latest version on https://rubygems.org/gems/dependabot-core, then that's a lot easier to understand than if we've got a version here in core that is on `0.200'ish` but rubygems is still way back on `0.95.1`... Obviously we could then publish the new one, but they'll still inevitably drift over time since we don't plan to publish this gem on every release of `dependabot-core`.

So as the versions drift, I start to wonder if we simply haven't published the gem and wonder how much drift there's been in the code from what's on rubygems. So making it clear they're in sync via the hardcoded version seems like the least confusing option for now.

So I'm going to merge as-is for now.

Obviously if we ever start using this gem, we should immediately switch to using `Dependabot::VERSION`.